source/adapters/cuda/usm.cpp

```diff
@@ -396,40 +465,58 @@ ur_usm_pool_handle_t_::ur_usm_pool_handle_t_(ur_context_handle_t Context,
     pNext = BaseDesc->pNext;
   }
 
-  auto MemProvider =
-      umf::memoryProviderMakeUnique<USMHostMemoryProvider>(Context, nullptr)
-          .second;
-
-  HostMemPool =
-      umf::poolMakeUnique<usm::DisjointPool, 1>(
-          {std::move(MemProvider)},
-          this->DisjointPoolConfigs.Configs[usm::DisjointPoolMemType::Host])
-          .second;
+  ur_result_t Ret;
+  std::tie(Ret, PoolManager) =
+      usm::pool_manager<usm::pool_descriptor>::create();
+  if (Ret) {
+    throw UsmAllocationException(Ret);
+  }
 
   auto Device = Context->DeviceID;
-  MemProvider =
-      umf::memoryProviderMakeUnique<USMDeviceMemoryProvider>(Context, Device)
-          .second;
-  DeviceMemPool =
-      umf::poolMakeUnique<usm::DisjointPool, 1>(
-          {std::move(MemProvider)},
-          this->DisjointPoolConfigs.Configs[usm::DisjointPoolMemType::Device])
-          .second;
-
-  MemProvider =
-      umf::memoryProviderMakeUnique<USMSharedMemoryProvider>(Context, Device)
-          .second;
-  SharedMemPool =
-      umf::poolMakeUnique<usm::DisjointPool, 1>(
-          {std::move(MemProvider)},
-          this->DisjointPoolConfigs.Configs[usm::DisjointPoolMemType::Shared])
-          .second;
+  auto UrUSMPool = reinterpret_cast<ur_usm_pool_handle_t>(this);
+
+  // TODO: Replace this with appropriate usm::pool_descriptor 'create' static
+  // function.
+  usm::pool_descriptor Descs[] = {
+      {UrUSMPool, Context, nullptr, UR_USM_TYPE_HOST, false},
+      {UrUSMPool, Context, Device, UR_USM_TYPE_DEVICE, false},
+      {UrUSMPool, Context, Device, UR_USM_TYPE_SHARED, false}};
+
+  // Helper lambda function matching USM type to DisjointPoolMemType
+  auto descTypeToDisjointPoolType =
+      [](usm::pool_descriptor &Desc) -> usm::DisjointPoolMemType {
+    switch (Desc.type) {
+    case UR_USM_TYPE_HOST:
+      return usm::DisjointPoolMemType::Host;
+    case UR_USM_TYPE_DEVICE:
+      return usm::DisjointPoolMemType::Device;
+    case UR_USM_TYPE_SHARED:
+      return (Desc.deviceReadOnly) ? usm::DisjointPoolMemType::SharedReadOnly
+                                   : usm::DisjointPoolMemType::Shared;
+    default:
+      // Added to suppress 'not all control paths return a value' warning.
```

Comment:
could put an `unreachable` here like this https://github.com/oneapi-src/unified-runtime/blob/39074a8eb604cc1a74180002719dadb3699fd778/source/adapters/cuda/usm.cpp#L258

Reply:
Done.